app/views/samples/_form_assays_js.haml

```diff
@@ -3,19 +3,28 @@
     dropzoneOptions = {
       paramName: "assay_files",
       autoProcessQueue: true,
+      maxThumbnailFilesize: 20,
       parallelUploads: 10,
       url: '/assay_files/create',
       uploadMultiple: true,
+      renameFile: function (file) {
+        var completeFileName = file.name;
+        var sanitizedName = completeFileName.replace(/[^a-z^A-Z^0-9\^_.-]/g, '_');
```

Comment:
This is _very_ restrictive, just a few ASCII characters. As long as CDx operates only in English speaking countries that should be acceptable, but it will be a problem in other languages, like Spanish or [Sinhala](https://en.wikipedia.org/wiki/Sinhala_language) (Sri Lanka).

From my experience we don't need to sanitize much characters (see this [wikipedia article](https://en.wikipedia.org/wiki/Filename#Reserved_characters_and_words)). I think the following would be nicer internationally. It should accepts all unicode letters & numbers, whatever the language (I didn't know `\p{}` before, it's _very_ nice):

```js
/[^\p{Alpha}\p{Numeric}_.-]/gu
```

Also, it's better to store the file with a completely different name (e.g. as a UUID, like `c7b78085-33b4-4a93-929c-9358ab1b2952.pdf`) then set the `Content-Disposition` header when downloading to set the original filename (using the `filename*=UTF-8''` form). Then you wouldn't have to bother sanitizing the filename.